website/content/api-docs/secret/terraform.mdx

```diff
@@ -0,0 +1,310 @@
+
+- `max_ttl` `(duration: "")` – Specifies the max TTL for this role. This is provided
+  as a string duration with a time suffix like `"30s"` or `"1h"` or as seconds. If not
+  provided, the default Vault Max TTL is used. Only applies to User API tokens.
+
+
+### Sample Payload
+
+To create a Vault role to manage a Terraform Cloud User tokens
+
+```json
+{
+  "user_id": "user-glhf1234",
+  "ttl":"1h",
+  "max_ttl":"24h",
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request POST \
+    --header "X-Vault-Token: ..." \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/terraform/role/tfuser
+```
+
+## Read Role
+
+This endpoint queries for information about a Terraform Cloud role with the given name.
+If no role exists with that name, a 404 is returned.
+
+| Method | Path                |
+| :----- | :------------------ |
+| `GET`  | `/terraform/role/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the role to query. This
+  is part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/terraform/role/tfuser
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "max_ttl": 86400,
+    "name": "tfuser",
+    "ttl": 3600,
+    "user_id": "user-glhf1234"
+  },
+}
+```
+
+## List Roles
+
+This endpoint lists all existing roles in the backend.
+
+| Method | Path                    |
+| :----- | :---------------------- |
+| `LIST` | `/terraform/role`           |
+| `GET`  | `/terraform/role?list=true` |
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request LIST \
+    http://127.0.0.1:8200/v1/terraform/role
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "keys": ["tfuser"]
+  }
+}
+```
+
+## Delete Role
+
+This endpoint deletes a Terraform Cloud role with the given name. Even if the role does
+not exist, this endpoint will still return a successful response.
+
+| Method   | Path                |
+| :------- | :------------------ |
+| `DELETE` | `/terraform/role/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the role to delete. This
+  is part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request DELETE \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/terraform/role/tfuser
+```
+
+## Rotate Role
+
+This endpoint rotates the credentials for a Terraform Cloud role that manages an
+Organization or Team. 
+
+
+| Method   | Path                |
+| :------- | :------------------ |
+| `POST` | `/terraform/rotate-role/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the role to rotate. This
+  is part of the request URL.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request POST \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/terraform/rotate-role/testing
+```
+
+## Generate Credential
```

Comment:
nit: Would this be better titled as "Generate token"?

Reply:
I _believe_ `Credential` is consistent with other secret engines, so unless you strongly object I'd like to leave as is